inc/search.class.php

```diff
@@ -4867,7 +4868,7 @@ static function addLeftJoin($itemtype, $ref_table, array &$already_link_tables,
 
       // Do not take into account standard linkfield
       $tocheck = $nt.".".$linkfield;
-      if ($linkfield == getForeignKeyFieldForTable($new_table)) {
+      if ($linkfield == $ref_fk || strpos($linkfield, $ref_fk . '_') === 0) {
```

Comment:
```suggestion
      if (getTableNameForForeignKeyField($linkfield) !== $new_table) {
```

Reply:
Sorry, my suggestion was wrong (`!=` VS `==`).
```suggestion
      if (getTableNameForForeignKeyField($linkfield) == $new_table) {
```